db/compaction/compaction_picker_fifo.cc

```diff
@@ -79,10 +79,16 @@ Compaction* FIFOCompactionPicker::PickTTLCompaction(
       FileMetaData* f = *ritr;
```

Comment:
Based on my reading of the docs https://github.com/facebook/rocksdb/wiki/Time-to-Live, it seems like we will want to use `newest_key_time`, since we want to "guarantees that key-values inserted will remain in the db for _at least_ ttl amount of time"

If this is true, I will make another update to this PR to change the log message a few lines down ("FIFO compaction: picking file...")

Reply:
The `creation_time` field is recorded during flush (https://github.com/facebook/rocksdb/blob/ddafba870d261878e151775c300674357ddb74e7/db/builder.cc#L324), which is roughly the same as the newest key time. It doesn't work well after a file is compacted and this PR fixes that.

Reply:
Ah gotcha, I see this code is for level 0 files. That makes sense.